src/backend/gl/src/window/glutin.rs

```diff
@@ -118,33 +210,93 @@ impl Surface {
 
 impl hal::Surface<B> for Surface {
     fn kind(&self) -> hal::image::Kind {
-        let ex = get_window_extent(&self.window);
-        let samples = self.window
-            .get_pixel_format()
-            .multisampling
-            .unwrap_or(1);
-        hal::image::Kind::D2(ex.width, ex.height, 1, samples as _)
+        match self {
+            Surface::Swapchained(surface) => {
+                let extent = physical_to_extent(
+                    &surface
+                    .surface_context
+                    .window
+                    .get_inner_size()
+                    .unwrap()
+                    .to_physical(surface.surface_context.window.get_hidpi_factor())
+                );
+                let samples = surface
+                    .surface_context
+                    .context
+                    .get_pixel_format()
+                    .multisampling
+                    .unwrap_or(1);
+                hal::image::Kind::D2(extent.width, extent.height, 1, samples as _)
+            }
+            _ => unimplemented!(),
+        }
     }
 
+    // TODO return proper requirments, currently doesn't matter because we just
```

Comment:
I personally don't think addressing this is relevant to this PR, but that's debatable.